### ISSUE TITLE
For logged in users with "Edit site" permisson, allow full access to disabled sites

### ISSUE THREAD

**Issue by mmikkel** [craftcms/cms#7265]
### Description

Authoring and previewing content in a disabled site is currently (Craft 3.5.16) quite cumbersome.

Fixing #7264 would help, but it'd also be great if Craft could effectively ignore the disabled "Enable on the frontend" site setting, if the site is being browsed by a logged-in user that has the `editSite:[SiteUID]` permission for that disabled site (or maybe a new permission – `viewDisabledSite:[SiteUID]` or similar – that permits viewing disabled sites without needing a `siteToken` param in the request URL).

**Comment by brandonkelly** [comment]
I’m not totally sure about this one. It could lead to confusion by admins, who could think that the “Enable on the front end” setting isn’t actually working.

**Comment by mmikkel** [comment]
@brandonkelly Yeah, I guess it comes down to how admins expect that setting to work... is it comparable to disabling an entry, or is it more like a per-site `isSystemLive`? I guess it's actually more like the former, in which case I agree that this FR could be confusing.

What sparked this FR is the use case where we've added a new site to an existing multi-site, and that site needs to be inaccessible to visitors while authors spend time adding content to it. Disabling the site makes the authoring workflow really difficult, since the only way to view content is via preview.

For now, I've opted to leave the site enabled, and to set the `isSystemLive` setting dynamically based on the current site/logged-in user (which is what I've done in similar situations, pre 3.5). But, if there was a way to add something like a per-site `isSystemLive` in core, that'd be great!
